Continue porting WebKit/ related types to the new IPC serialization format

gavin-apple opened the pull request:
#### 3168f2acf3cb7416bbe4af15c1285896f2367144
<pre>
Continue porting WebKit/ related types to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=252742">https://bugs.webkit.org/show_bug.cgi?id=252742</a>
rdar://105776889

Reviewed by Alex Christensen.

This change includes porting across the following types to the new
serialization format:
    - GamepadData
    - WebAutocorrectionData
    - NavigationActionData
    - PlatformPopupMenuData
    - ItemType
    - TouchBarMenuItemData
    - WebsiteData::Entry
    - WebsiteData
    - WebsiteDataType

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::hasLocalStorage):
(WebKit::NetworkProcess::registrableDomainsWithWebsiteData):
(WebKit::NetworkProcess::websiteDataOriginDirectoryForTesting):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/NetworkProcess.messages.in:
* Source/WebKit/Shared/Gamepad/GamepadData.cpp:
(WebKit::GamepadData::GamepadData):
(WebKit::GamepadData::encode const): Deleted.
(WebKit::GamepadData::decode): Deleted.
* Source/WebKit/Shared/Gamepad/GamepadData.h:
* Source/WebKit/Shared/Gamepad/GamepadData.serialization.in: Added.
* Source/WebKit/Shared/NavigationActionData.cpp: Removed.
* Source/WebKit/Shared/NavigationActionData.h:
* Source/WebKit/Shared/NavigationActionData.serialization.in: Added.
* Source/WebKit/Shared/PlatformPopupMenuData.cpp: Removed.
* Source/WebKit/Shared/PlatformPopupMenuData.h:
* Source/WebKit/Shared/PlatformPopupMenuData.serialization.in: Added.
* Source/WebKit/Shared/TouchBarMenuItemData.cpp:
(WebKit::TouchBarMenuItemData::TouchBarMenuItemData):
(WebKit::TouchBarMenuItemData::encode const): Deleted.
(WebKit::TouchBarMenuItemData::decode): Deleted.
* Source/WebKit/Shared/TouchBarMenuItemData.h:
* Source/WebKit/Shared/TouchBarMenuItemData.serialization.in: Added.
* Source/WebKit/Shared/WebsiteData/WebsiteData.cpp:
(WebKit::WebsiteData::Entry::Entry):
(WebKit::WebsiteData::Entry::isolatedCopy const):
(WebKit::WebsiteData::Entry::isolatedCopy):
(WebKit::WebsiteData::Entry::encode const): Deleted.
(WebKit::WebsiteData::Entry::decode): Deleted.
(WebKit::WebsiteData::encode const): Deleted.
(WebKit::WebsiteData::decode): Deleted.
* Source/WebKit/Shared/WebsiteData/WebsiteData.h:
(WebKit::WebsiteData::Entry::type const):
* Source/WebKit/Shared/WebsiteData/WebsiteData.serialization.in: Added.
* Source/WebKit/Shared/WebsiteData/WebsiteDataType.h:
* Source/WebKit/Shared/WebsiteData/WebsiteDataType.serialization.in: Added.
* Source/WebKit/Shared/ios/WebAutocorrectionData.h:
* Source/WebKit/Shared/ios/WebAutocorrectionData.mm: Removed.
* Source/WebKit/Shared/ios/WebAutocorrectionData.serialization.in: Added.
* Source/WebKit/Sources.txt:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
(-[WKWebsiteDataStore _originDirectoryForTesting:topOrigin:type:completionHandler:]):
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::websiteDataOriginDirectoryForTesting):
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::fetchDataAndApply):
(WebKit::WebsiteDataStore::originDirectoryForTesting):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/260751@main">https://commits.webkit.org/260751@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0dbce18595e10dc47aaaba82e9d409dc3db5f722

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109219 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18303 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42032 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/746 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118449 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19762 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9604 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101462 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114975 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14814 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98046 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42986 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96786 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29700 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84735 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11076 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31043 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11802 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7976 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17171 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50646 "Passed tests") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/74/builds/7421 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13425 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/WebsiteData/WebsiteData.h:
```diff
@@ -44,15 +44,17 @@ enum class WebsiteDataProcessType { Network, UI, Web };
 
-        void encode(IPC::Encoder&) const;
-        static std::optional<WebsiteData::Entry> decode(IPC::Decoder&);
+        WebCore::SecurityOriginData m_origin;
```
This is a struct, so don't add the m_ prefix to the variable names.

achristensen07 commented on Source/WebKit/NetworkProcess/NetworkProcess.cpp:
```diff
@@ -2350,13 +2350,15 @@ void NetworkProcess::renameOriginInWebsiteData(PAL::SessionID sessionID, Securit
-void NetworkProcess::websiteDataOriginDirectoryForTesting(PAL::SessionID sessionID, ClientOrigin&& origin, WebsiteDataType dataType, CompletionHandler<void(const String&)>&& completionHandler)
+void NetworkProcess::websiteDataOriginDirectoryForTesting(PAL::SessionID sessionID, ClientOrigin&& origin, OptionSet<WebsiteDataType> dataType, CompletionHandler<void(const String&)>&& completionHandler)
 {
+    RELEASE_ASSERT(dataType.hasExactlyOneBitSet());
```
This seems a bit harsh.  We could just assert not reached and early return.

gavin-apple committed:
Continue porting WebKit/ related types to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=252742
rdar://105776889

Reviewed by Alex Christensen.

This change includes porting across the following types to the new
serialization format:
    - GamepadData
    - WebAutocorrectionData
    - NavigationActionData
    - PlatformPopupMenuData
    - ItemType
    - TouchBarMenuItemData
    - WebsiteData::Entry
    - WebsiteData
    - WebsiteDataType

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::hasLocalStorage):
(WebKit::NetworkProcess::registrableDomainsWithWebsiteData):
(WebKit::NetworkProcess::websiteDataOriginDirectoryForTesting):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/NetworkProcess.messages.in:
* Source/WebKit/Shared/Gamepad/GamepadData.cpp:
(WebKit::GamepadData::GamepadData):
(WebKit::GamepadData::encode const): Deleted.
(WebKit::GamepadData::decode): Deleted.
* Source/WebKit/Shared/Gamepad/GamepadData.h:
* Source/WebKit/Shared/Gamepad/GamepadData.serialization.in: Added.
* Source/WebKit/Shared/NavigationActionData.cpp: Removed.
* Source/WebKit/Shared/NavigationActionData.h:
* Source/WebKit/Shared/NavigationActionData.serialization.in: Added.
* Source/WebKit/Shared/PlatformPopupMenuData.cpp: Removed.
* Source/WebKit/Shared/PlatformPopupMenuData.h:
* Source/WebKit/Shared/PlatformPopupMenuData.serialization.in: Added.
* Source/WebKit/Shared/TouchBarMenuItemData.cpp:
(WebKit::TouchBarMenuItemData::TouchBarMenuItemData):
(WebKit::TouchBarMenuItemData::encode const): Deleted.
(WebKit::TouchBarMenuItemData::decode): Deleted.
* Source/WebKit/Shared/TouchBarMenuItemData.h:
* Source/WebKit/Shared/TouchBarMenuItemData.serialization.in: Added.
* Source/WebKit/Shared/WebsiteData/WebsiteData.cpp:
(WebKit::WebsiteData::Entry::Entry):
(WebKit::WebsiteData::Entry::isolatedCopy const):
(WebKit::WebsiteData::Entry::isolatedCopy):
(WebKit::WebsiteData::Entry::encode const): Deleted.
(WebKit::WebsiteData::Entry::decode): Deleted.
(WebKit::WebsiteData::encode const): Deleted.
(WebKit::WebsiteData::decode): Deleted.
* Source/WebKit/Shared/WebsiteData/WebsiteData.h:
(WebKit::WebsiteData::Entry::type const):
* Source/WebKit/Shared/WebsiteData/WebsiteData.serialization.in: Added.
* Source/WebKit/Shared/WebsiteData/WebsiteDataType.h:
* Source/WebKit/Shared/WebsiteData/WebsiteDataType.serialization.in: Added.
* Source/WebKit/Shared/ios/WebAutocorrectionData.h:
* Source/WebKit/Shared/ios/WebAutocorrectionData.mm: Removed.
* Source/WebKit/Shared/ios/WebAutocorrectionData.serialization.in: Added.
* Source/WebKit/Sources.txt:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
(-[WKWebsiteDataStore _originDirectoryForTesting:topOrigin:type:completionHandler:]):
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::websiteDataOriginDirectoryForTesting):
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::fetchDataAndApply):
(WebKit::WebsiteDataStore::originDirectoryForTesting):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/260751@main
